internal/controller/controllers/infraenv_controller.go

```diff
@@ -660,6 +683,13 @@ func (r *InfraEnvReconciler) updateInfraEnvStatus(
 		isoUpdated = true
 	}
 
+	if internalInfraEnv.StaticNetworkConfig != "" && internalInfraEnv.ID != nil {
```

Comment:
1. How would internalInfraEnv.ID be nil?
2. You should check if infraEnv.Status.StaticNetworkDownloadURL is empty, otherwise you'll generate the same URL on every reconcile